callbacks.py

```diff
@@ -533,6 +534,85 @@ def _compute_drone_time(
         indicator_graphic_3, indicator_graphic_4, indicator_graphic_1, indicator_graphic_1
 
 
+@app.callback(
+    Output('hash', 'value'),
+    [Input('seq_start', 'n_clicks'), Input('app-tabs', 'active_tab')],
+    [State('input_drone', 'value'),
+     State('upload-starting-points', 'contents'),
+     State('upload-incidents', 'contents'),
+     State('speed', 'value'),
+     State('acc', 'value'),
+     State('vert-acc', 'value'),
+     State('alt', 'value'),
+     State('dep_delay', 'value'),
+     State('arr_delay', 'value'),
+     State('detec_delay', 'value'),
+     State('day', 'value'),
+     State('detec_rate_home', 'value'),
+     State('wit_detec', 'value'),
+     State('detec_rate_vp', 'value'),
+     State('unavail_delta', 'value'),
+     State('lang', 'value'),
+     State('hash', 'value')])
+def params_hash(unused_seq_start, unused_active_tab, *args):
+    return _compute_params_hash(*args)
+
+
+@app.callback(
+    Output('hash_b', 'value'),
+    [Input('seq_start_b', 'n_clicks'), Input('app-tabs', 'active_tab')],
+    [State('input_drone_b', 'value'),
+     State('upload-starting-points', 'contents'),
+     State('upload-incidents', 'contents'),
+     State('speed_b', 'value'),
+     State('acc_b', 'value'),
+     State('vert-acc_b', 'value'),
+     State('alt_b', 'value'),
+     State('dep_delay_b', 'value'),
+     State('arr_delay_b', 'value'),
+     State('detec_delay_b', 'value'),
+     State('day_b', 'value'),
+     State('detec_rate_home_b', 'value'),
+     State('wit_detec_b', 'value'),
+     State('detec_rate_vp_b', 'value'),
+     State('unavail_delta_b', 'value'),
+     State('lang', 'value'),
+     State('hash_b', 'value')])
+def params_hash_b(unused_seq_start, unused_active_tab, *args):
+    return _compute_params_hash(*args)
+
+
+def _compute_params_hash(
+        drone_input, custom_drone_input, custom_incidents_csv,
+        input_speed, input_acc, vert_acc, alt, dep_delay, arr_delay, detec_delay,
+        input_jour, detec_rate_home, no_witness_rate, detec_rate_vp, unavail_delta, lang,
+        previous_hash):
+    combined = hashlib.sha1()
```

Comment:
I get an error because detec_rate_home, no_witness_rate and detec_rate_vp are integers, therefore cannot be encoded. I suggest the following addition there : 

    detec_rate_home = str(detec_rate_home)  
    no_witness_rate = str(no_witness_rate)
    detec_rate_vp = str(detec_rate_vp)

Reply:
You're right, I'm not sure why it didn't trigger when I tried it. 

I've updated the code to stay on the safe side for all vars.